### ISSUE TITLE
Issue running 3DMark DXR test Port Royal..

### ISSUE THREAD

**Issue by oscarbg** [HansKristian-Work/vkd3d-proton#602]
Hi,
just running 3DMark Port Royal DXR from Windows..
fails upon loading and closes..
log using VKD3D_DEBUG=trace but still not using a VKD3D debug build (will post later) points to:
1036:err:vkd3d_dxil_log_callback: dxil-spirv: Invalid type ID in cast<T>.
full log:
[portroyalvkd3dlog.txt](https://github.com/HansKristian-Work/vkd3d-proton/files/6217103/portroyalvkd3dlog.txt)

note: 3DMark also includes a DXR feature test.. but this test requires DXR 1.1 (uses ray queries?) and not sure if DXR 1.1 supported by VKD3D yet..

**Comment by SveSop** [comment]
> note: 3DMark also includes a DXR feature test.. but this test requires DXR 1.1 (uses ray queries?) and not sure if DXR 1.1 supported by VKD3D yet..

DXR 1.1 is not yet supported.

**Comment by oscarbg** [comment]
@SveSop yep I know..

just pointing that tested with latest master to see if:
`1036:err:vkd3d_dxil_log_callback: dxil-spirv: Invalid type ID in cast<T>.`
had been fixed by latest cast fixes but seems I updated to latest 3DMark and now the failing is different it fails before showing loading screen (i.e. earlier) and the culprit seems to be d3d12_device_CreateSharedHandle being a stub:
```
19820:fixme:d3d12_device_CreateSharedHandle: iface 000000001392c040, object 000000001d1ad4a0, attributes 0000000000000000, access 0x10000000, name "Intermediate copy texture 0", handle 000000000014d9b0 stub!
19820:fixme:d3d12_device_CreateSharedHandle: iface 000000001392c040, object 000000001d1ad4a0, attributes 0000000000000000, access 0x10000000, name "Intermediate copy texture 0", handle 000000000014d9b0 stub!
```

full log:
[vkd3dlog.txt](https://github.com/HansKristian-Work/vkd3d-proton/files/6487504/vkd3dlog.txt)

**Comment by Joshua-Ashton** [comment]
This will not be implemented any time soon, it needs Wine work for shared resource shite.

**Comment by oscarbg** [comment]
seeing now 3DMark night raid and time spy render OK with VKD3D, next seems Port Royal..
@HansKristian-Work @Joshua-Ashton @Guy1524 curious how difficult is to support shared resources in VKD3D seeing up to date wine forks with support for shared resources (https://github.com/Guy1524/wine/commits/shared-resources)
thanks..

**Comment by Etaash-mathamsetty** [comment]
Does it work now with the latest proton ge or proton experimental?

**Comment by Blisto91** [comment]
@oscarbg Friendly ping. Would you be able to retest this?

**Comment by MegWATTT** [comment]
@oscarbg 
> ```
> 19820:fixme:d3d12_device_CreateSharedHandle: iface 000000001392c040, object 000000001d1ad4a0, attributes 0000000000000000, access 0x10000000, name "Intermediate copy texture 0", handle 000000000014d9b0 stub!
> 19820:fixme:d3d12_device_CreateSharedHandle: iface 000000001392c040, object 000000001d1ad4a0, attributes 0000000000000000, access 0x10000000, name "Intermediate copy texture 0", handle 000000000014d9b0 stub!
> ```

Make sure you only have one vulkan driver reported (llvmpipe doesn't count, but having both RADV and AMDVLK installed count as 2 drivers). It looks like 3DMark's DX12 demos will use shared resources in the case they think there is more than 1 device. I use the 'DXVK_FILTER_DEVICE_NAME' environment variable to do this.

As for Port Royal, for some reason, I'm unable to start the benchmark on Linux. It may be a Wine/Proton issue as I'm able to get to the loading screen on Windows. However, it crashes at the end of the loading. The log shows the same issue as the first post: ending with `1036:err:vkd3d_dxil_log_callback: dxil-spirv: Invalid type ID in cast<T>.`


For the 3DMark DXR test, I can't start the bench on Proton too. However, this one works fine on Windows, with lower performance (~12 FPS on VKD3D instead of ~16)

Tested on Proton Experimental, VKD3D 2.8, mesa 23.0, RX 6700, with VKD3D_CONFIG=dxr11 and VKD3D_FEATURE_LEVEL=12_2

**Comment by oscarbg** [comment]
Yep will try to test that soon.. my RTX GPU died some months ago and using a Maxwell since then..

**Comment by Etaash-mathamsetty** [comment]
> @oscarbg
> 
> > ```
> > 19820:fixme:d3d12_device_CreateSharedHandle: iface 000000001392c040, object 000000001d1ad4a0, attributes 0000000000000000, access 0x10000000, name "Intermediate copy texture 0", handle 000000000014d9b0 stub!
> > 19820:fixme:d3d12_device_CreateSharedHandle: iface 000000001392c040, object 000000001d1ad4a0, attributes 0000000000000000, access 0x10000000, name "Intermediate copy texture 0", handle 000000000014d9b0 stub!
> > ```
> 
> Make sure you only have one vulkan driver reported (llvmpipe doesn't count, but having both RADV and AMDVLK installed count as 2 drivers). It looks like 3DMark's DX12 demos will use shared resources in the case they think there is more than 1 device. I use the 'DXVK_FILTER_DEVICE_NAME' environment variable to do this.
> 
> As for Port Royal, for some reason, I'm unable to start the benchmark on Linux. It may be a Wine/Proton issue as I'm able to get to the loading screen on Windows. However, it crashes at the end of the loading. The log shows the same issue as the first post: ending with `1036:err:vkd3d_dxil_log_callback: dxil-spirv: Invalid type ID in cast<T>.`
> 
> For the 3DMark DXR test, I can't start the bench on Proton too. However, this one works fine on Windows, with lower performance (~12 FPS on VKD3D instead of ~16)
> 
> Tested on Proton Experimental, VKD3D 2.8, mesa 23.0, RX 6700, with VKD3D_CONFIG=dxr11 and VKD3D_FEATURE_LEVEL=12_2

you also have to do VKD3D_FILTER_DEVICE_NAME for dx12 benchmarks to work and also install native d3dcompiler_43 and d3dcompiler_47 in your prefix

**Comment by oscarbg** [comment]
just testing from Windows and seems the bug 
err:vkd3d_dxil_log_callback: dxil-spirv: Invalid type ID in cast<T>.
is still present..

**Comment by Blisto91** [comment]
@oscarbg Could i get you to give this one another spin?

**Comment by MarijnS95** [comment]
Moved to https://github.com/HansKristian-Work/vkd3d-proton/issues/2809. @Etaash-mathamsetty I would recommend only quoting the relevant part of my rather large comment, and removing the accidental duplicate reply 😉
